### PR TITLE
Do not check for PK of project layers if not a vector layer

### PR DESCRIPTION
**Pull request by suricactus**
Fixing "OpenStreetMap does not have a PK".

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-2892 Show information when the project will have some layers as readonly](https://app.clickup.com/t/2192114/QF-2892)

**Review by nirvn — APPROVED** [review]

File: docker-app/qfieldcloud/core/models.py [hunk 1]
```diff
@@ -1219,22 +1219,23 @@ def problems(self) -> list[dict]:
...
+                                "layer": layer_name,
+                                "level": "warning",
+                                "code": "layer_problem",
+                                "description": _(
+                                    'Layer "{}" does not have supported primary key attribute. The layer will be read-only on QField.'
+                                ).format(
+                                    layer_name,
+                                ),
+                                "solution": _(
+                                    "To make the layer editable on QField, store the layer data in a GeoPackage or PostGIS layer with single column primary key."
```
Comment (why-not-try-calmer) [inline docker-app/qfieldcloud/core/models.py:1235]:
```suggestion
                                    "To make the layer editable on QField, store the layer data in a GeoPackage or PostGIS layer, using a single column as primary key."
```

File: docker-app/qfieldcloud/core/models.py [hunk 2]
```diff
@@ -1219,22 +1219,23 @@ def problems(self) -> list[dict]:
...
+                elif layer_data.get("layer_type_name") in ("VectorLayer", "Vector"):
+                    if layer_data.get("qfc_source_data_pk_name") == "":
+                        problems.append(
+                            {
+                                "layer": layer_name,
+                                "level": "warning",
+                                "code": "layer_problem",
+                                "description": _(
+                                    'Layer "{}" does not have supported primary key attribute. The layer will be read-only on QField.'
+                                ).format(
```
Comment (why-not-try-calmer) [inline docker-app/qfieldcloud/core/models.py:1231]:
```suggestion
                                    'Layer "{}" does not support the `primary key` attribute. The layer will be read-only on QField.'
                                ).format(
```

**Review by why-not-try-calmer — CHANGES_REQUESTED** [review]
Just a couple of suggestions.

**Review by why-not-try-calmer — APPROVED** [review]

**Pull request merged**